llrt_core/src/vm.rs

```diff
@@ -630,14 +631,19 @@ fn init(ctx: &Ctx<'_>, module_names: HashSet<&'static str>) -> Result<()> {
 
             let rt = unsafe { qjs::JS_GetRuntime(ctx.as_raw().as_ptr()) };
 
-            let mut executing_timers: Option<Vec<Option<ExecutingTimer>>> = Some(Vec::new());
+            let mut deadline = Instant::now();
+
+            let mut executing_timers = Vec::new();
 
             let imported_object = loop {
                 if let Some(x) = import_promise.result::<Object>() {
                     break x?;
                 }
 
-                poll_timers(rt, &mut executing_timers, &mut false);
+                if deadline < Instant::now() {
+                    poll_timers(rt, &mut executing_timers, None, Some(&mut deadline));
```

Comment:
Is that not conflicting with your background task? Why is it still needed?

Reply:
It's not conflicting because require is synchronous and might load a file containing something like:
```javascript
await new Promise((res) => setTimeout(res, 0));
```
Which never will be resolved since we're not polling the futures yet which happens in here: https://github.com/awslabs/llrt/blob/6cbad7193b68e0c01fc893cd357d84ea0a243215/llrt_core/src/vm.rs#L489